policyengine_api/services/policy_service.py

```diff
@@ -18,3 +59,134 @@ def get_policy_json(self, country_id, policy_id):
+            # Check if policy already exists
+            print("Checking if policy exists")
+            existing_policy = self._get_unique_policy_with_label(
+                country_id, policy_hash, label
+            )
+
+            # If so, pass appropriate values back
+            if existing_policy:
+                print("Policy already exists")
+                existing_policy_id = str(existing_policy["id"])
+                message = (
+                    "Warning: Record created previously with this label. To create "
+                    + "a new record, change the submitted data's country ID, policy "
+                    + "parameters, or label, and emit the request again"
+                )
+                return existing_policy["id"], "Policy already exists", True
+
+            # Otherwise, insert the new policy...
+            print("Policy does not exist; creating new policy")
+            self._create_new_policy(
+                country_id, policy_json, policy_hash, label, api_version
+            )
+
+            # And then fetch it back out; once an ORM is added, we can
+            # just return policy ID directly from the insert
+            new_policy = self._get_unique_policy_with_label(
+                country_id, policy_hash, label
+            )
+
+            return int(new_policy["id"]), "Policy created", False
+
+        except Exception as e:
+            print(f"Error setting policy: {str(e)}")
+            raise e
+
+    def _create_new_policy(
+        self,
+        country_id: str,
+        policy_json: dict,
+        policy_hash: str,
+        label: str | None,
+        api_version: str,
+    ) -> None:
+        """
+        Insert a new policy into the database
+
+        Arguments
+            country_id: str -- the country ID
+            policy_json: dict -- the policy content
+            policy_hash: str -- the policy content hash
+            label: str | None -- the policy label
+            api_version: str -- the API version
+
+        """
+        try:
+            database.query(
+                f"INSERT INTO policy (country_id, policy_json, policy_hash, label, api_version) VALUES (?, ?, ?, ?, ?)",
+                (
+                    country_id,
+                    json.dumps(policy_json),
+                    policy_hash,
+                    label,
+                    api_version,
+                ),
+            )
+        except Exception as e:
+            print(f"Error creating new policy: {str(e)}")
+            raise e
+
+    def _get_unique_policy_with_label(
+        self, country_id: str, policy_hash: str, label: str
+    ) -> dict | None:
+        """
+        Given policy content (represented as a hash) and a label, fetch the policy;
+        this method ensures that both policy content and label are unique, a workaround
+        to an old issue whereby multiple copies of the same policy/label pair could be created
+
```

Comment:
**accolade and very nitty-nonblocking comment** This bit of the comment is a great example of useful information not obvious from the naming (and not appropriate to add to it).

I'd still get rid of the rest of it though